src/super_gradients/training/models/detection_models/yolo_base.py

```diff
@@ -205,7 +205,12 @@ def __init__(
         self.n_anchors = 1
         self.grid = [torch.zeros(1)] * self.detection_layers_num  # init grid
 
-        self.register_buffer("stride", torch.tensor(stride), persistent=False)
+        if torch.is_tensor(stride):
+            stride = stride.clone().detach()
+        else:
+            stride = torch.tensor(stride)
+
+        self.register_buffer("stride", stride, persistent=False)
```

Comment:
Is this just optimization, or it also meant to fix the bug/another bug?
I think that the only noticeable difference is if stride is on GPU, `stride.clone().detach()` will create a copy on GPU, while `torch.tensor(stride)` will create a copy on CPU. Was that the motivation?

Reply:
Not exactly - this is no avoid a warning message if input is already a torch tensor.
Otherwise torch complains and suggest we use `x.clone().detach()` instead of `torch.tensor(x)`